src/libsyncengine/update_detection/file_system_observer/localfilesystemobserverworker.cpp

```diff
@@ -645,10 +648,16 @@ ExitInfo LocalFileSystemObserverWorker::exploreDir(const SyncPath &absoluteParen
             setExitCause(ExitCause::SyncDirAccesError);
             return {ExitCode::SystemError, ExitCause::SyncDirAccesError};
         }
-
         DirectoryEntry entry;
         bool endOfDirectory = false;
+        int perfMonitorCounter = 0; // We monitor the performance for the treatement of 1000 items
+        SentryHandler::ScopedPTrace perfMonitor(SentryHandler::PTraceName::LFSO_ExploreItem, syncDbId(), true);
         while (dirIt.next(entry, endOfDirectory, ioError) && !endOfDirectory && ioError == IoError::Success) {
+            perfMonitorCounter++;
+            if (perfMonitorCounter % 1000 == 0) {
+                perfMonitor.stopAndStart(SentryHandler::PTraceName::LFSO_ExploreItem, syncDbId(), true);
+            }
```

Comment:
Could we delegate this logic to the `SentryHandler::ScopedPTrace` class? The goal should always to add as less as possible into the caller method. `LocalFileSystemObserverWorker` is already quite complicated

Reply:
Quite nice 👍 However, it would be even better if it can be completely transparent on the caller side, meaning we always call `entryHandler::ScopedPTrace`. For example, a list of `PTraceName`could be maintained in `ScopedPTrace` for whom we will not send too many transactions.
Please, take may comment above as something "nice to have". Do not spend too much time optimizing, this is already nice

Reply:
Thank you for the suggestion. Unfortunately, I can't see a solution to achieve this without adding more complexity to ScopedPTrace, for which I've tried to maintain the simplest possible interface.
Let's discuss it this afternoon 👍🏾